integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -1169,3 +1182,69 @@ def do_it(spark):
 
     assert_gpu_and_cpu_are_equal_collect(do_it,
         conf={'spark.sql.ansi.enabled': 'true'})
+
+
+# Tests for standard deviation and variance aggregations.
+@approximate_float
+@ignore_order
+@incompat
+@pytest.mark.parametrize('data_gen', _init_list_with_nans_and_no_nans, ids=idfn)
+@pytest.mark.parametrize('conf', get_params(_confs, params_markers_for_confs), ids=idfn)
+def test_groupby_std_variance(data_gen, conf):
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark : gen_df(spark, data_gen, length=100),
+        "data_table",
+        'select ' +
+        'stddev(b),' +
+        'stddev_pop(b),' +
+        'stddev_samp(b),' +
+        'variance(b),' +
+        'var_pop(b),' +
+        'var_samp(b)' +
+        ' from data_table group by a',
+        conf=conf)
+
+
+# xfail because GPU-casting directly from decimal to double is not yet supported.
+# This test also includes the test above thus let remove the test above when this one can be enabled.
+@pytest.mark.xfail(reason='https://github.com/NVIDIA/spark-rapids/issues/3610')
```

Comment:
Bobby's fix went in, can we remove the xfail for this test?

Reply:
I did that and a test failed. Investigating...